CodeEdit/Features/AppPreferences/Model/Text Editing/TextEditingPreferences.swift

```diff
@@ -8,6 +8,31 @@
 import AppKit
 import Foundation
 
+extension String.Encoding: Codable {
+    static var allAscii: [String.Encoding] = [String.Encoding.ascii, String.Encoding.nonLossyASCII]
+    static var allJapanese: [String.Encoding] = [String.Encoding.iso2022JP, String.Encoding.japaneseEUC]
+    static var allLatin: [String.Encoding] = [String.Encoding.isoLatin1,
+                                              String.Encoding.isoLatin2,
+                                              String.Encoding.macOSRoman
+                                            ]
+    static var allUnicodes: [String.Encoding] = [String.Encoding.unicode,
+                                                 String.Encoding.utf8,
+                                                 String.Encoding.utf16,
+                                                 String.Encoding.utf16BigEndian,
+                                                 String.Encoding.utf16LittleEndian,
+                                                 String.Encoding.utf32,
+                                                 String.Encoding.utf32BigEndian,
+                                                 String.Encoding.utf32LittleEndian
+                                                ]
+    static var allWindows: [String.Encoding] = [String.Encoding.windowsCP1250,
+                                                String.Encoding.windowsCP1251,
+                                                String.Encoding.windowsCP1252,
+                                                String.Encoding.windowsCP1253,
+                                                String.Encoding.windowsCP1254
+                                                ]
+    static var others: [String.Encoding] = [String.Encoding.nextstep, String.Encoding.shiftJIS, String.Encoding.symbol]
```

Comment:
Please mind the [Code Style](https://github.com/CodeEditApp/CodeEdit/wiki/Code-Style#parameter-alignment)

```suggestion
    static var others: [String.Encoding] = [
        .nextstep, 
        .shiftJIS, 
        .symbol
    ]
```